File: javascript/rna-transcription/rna-transcription.js
const rna = {
  G: "C",
  C: "G",
  T: "A",
  A: "U"
};

export const toRna = dna =>
  dna
    .split("")
    .map(function(x) {
      if (x in rna) {
        return rna[x];
      } else {
        throw new Error("Invalid input");
      }
    })
    .join("");

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  RNA = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }.freeze

  def self.of_dna(dna)
    dna.chars.map { |char| RNA.fetch(char) { return '' } }.join
  end
end
